plugins/query-insights/src/main/java/org/opensearch/plugin/insights/core/exporter/QueryInsightExporter.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.plugin.insights.core.exporter;
+
+import org.opensearch.common.unit.TimeValue;
+import org.opensearch.plugin.insights.rules.model.SearchQueryRecord;
+
+import java.util.List;
+
+/**
+ * Simple abstract class to export data collected by search query analyzers
+ * <p>
+ * Mainly for use within the Query Insight framework
+ *
+ * @opensearch.internal
+ */
+public abstract class QueryInsightExporter<T extends SearchQueryRecord<?>> {
+
+    private boolean enabled = false;
+
+    /** The export interval of this exporter, default to 60 seconds */
+    private TimeValue exportInterval = TimeValue.timeValueSeconds(5);
```

Comment:
Comment says default is 60 seconds but code has 5 seconds? Also we can add a constant for this. Further, do we need to make this configurable somewhere?

Reply:
Good catch. I forgot to change this back after testing.